src/end_entity.rs

```diff
@@ -107,21 +108,21 @@ impl<'a> EndEntityCert<'a> {
     /// intermediate certificates that the server sent in the TLS handshake.
     /// `time` is the time for which the validation is effective (usually the
     /// current time).
-    pub fn verify_is_valid_cert_with_eku(
+    pub fn verify_for_usage(
         &self,
         supported_sig_algs: &[&SignatureAlgorithm],
-        &NonTlsTrustAnchors(trust_anchors): &NonTlsTrustAnchors,
+        trust_anchors: &[TrustAnchor],
         intermediate_certs: &[&[u8]],
         time: Time,
-        eku: KeyUsage,
+        usage: KeyUsage,
```

Comment:
nit: WDYT about updating the doc string to mention the `usage` and `crls` arguments? (_I think this was overlooked in earlier PRs, sorry!_).

Reply:
Good call, should definitely do that.